aten/src/ATen/cuda/CUDABlas.h

```diff
@@ -0,0 +1,251 @@
+#pragma once
+/*
+  Provides a subset of CUDA BLAS functions as templates:
+
+    gemm<Dtype>(stream, transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c,
+  ldc)
+
+    gemv<Dtype>(stream, transa, m, n, alpha, a, lda, x, incx, beta, y, incy)
+
+  where Dtype is double, float, or at::Half. The functions are
+  available in at::cuda::blas namespace.
+ */
+
+#include <ATen/cuda/CUDAContext.h>
+
+// In CUDA 8.0, definition of data types for sgemmex changed
+#if CUDA_VERSION < 8000
+#define CUDA_R_16F CUBLAS_DATA_HALF
+#endif
+
+#define TORCH_CUDABLAS_CHECK(EXPR)        \
+  do {                                    \
+    cublasStatus_t __err = EXPR;          \
+    if (__err != CUBLAS_STATUS_SUCCESS) { \
+      AT_ERROR(                           \
+          "CUDA error: ",                 \
+          _cublasGetErrorEnum(__err),     \
+          " when calling `" #EXPR "`");   \
+    }                                     \
+  } while (0)
+
+namespace {
+static const char* _cublasGetErrorEnum(cublasStatus_t error) {
```

Comment:
This appears to duplicate the functionality in `__THCublasCheck` and it doesn't look like that function was deleted. Can we delete the old function? (That can be a follow up PR.)

Reply:
An easy way to do this is to keep the old function name, but just have it call into the new implementation here. Even better, replace the old call sites to point to this new one.

Reply:
The port of CUDA Blas functions gemm and gemv is complete and deduplication of these in THCBlas.cu is resolved by calling `at::cuda::blas` functions.
For other CUDA Blas functions, I have created issue #21566 that reports the duplication of THCublasCheck and other functions. So, the issue can be resolved in a follow-up PR.